### ISSUE TITLE
Clarification request on "Automatic purge optimization" in Bambu Studio v1.10

### ISSUE THREAD

**Issue by TheRealHotFix** [bambulab/BambuStudio#5042]
I see in Bambu Studio BETA v01.10.00.74 the following information:

> 1. Purge optimization: Without color bleeding, official multicolor prints with some color filaments achieve an average of 20.9% purge savings.
> - Long retraction enabled for more materials: X/P/A series printer now supports long retraction for PLA Basic, PLA Matte, PLA Galaxy, PLA Marble, PLA Glow, PLA-CF, PETG HF, PETG Translucent, PETG-CF, ABS, ABS-GF, ASA, Support for PLA/PETG, and Support for ABS.
> - Purge optimization for 12 colors: Purge savings are now supported for Bambu PLA Basic filaments in 12 colors (Red/Black/Brown/Gray/Yellow/Bambu Green/Blue Gray/Light Gray/Dark Gray/Purple/Blue/Orange), significantly reducing waste in multicolor prints using these filaments. (Note: Automatic purge optimization is currently limited to Bambu Lab's official filaments. Due to the variations in third-party filaments, consistent color change results cannot be guaranteed. For third-party filaments, please adjust settings manually).

I'm trying to find out more about what exactly "automatic purge optimization" is. I went through the changes in the latest BETA from the last full release looking for the words "purge', "opt" (short for optimization), and filament. The only relevant hit I got was this one:
["ENH: enable long retractions for more filaments"](https://github.com/bambulab/BambuStudio/commit/d95781e244409df1b57cd59428482360c4a45503)
The text in that link as well as the JSON code tweaks speaks to information the first sub-bullet but not the second sub-bullet.

In regard to the second sub-bullet, I can't find any details on what exactly was done in respect to the 12 colors. Did the from/to values for use with the multiplier change? It sort of looks that way in the graphic, but I don't want to assume anything. If so, then how is that "automatic purge optimization" versus just optimizing the base multiplier values for specific filaments? I'm just trying to understand what exactly was done as I have been using the custom Gcode by LeonFisherSkipper for a while now, and I would like to switch over to the native solution(s) Bambu Studio, but that requires me understand what the native stuff is doing to continue to maintain my current print quality w/ reduced purge settings.

By the way, if there is a better way to ask a question about a change in a BETA (or even RTM) release, please let me know. I looked for a comment or discussion button on the release and didn't find any.

**Comment by TheRealHotFix** [comment]
I wanted to follow up on this now that more documentation has been released and close this out in case anyone else stumbles on it.

As best I can tell, the "automatic purge optimization" means someone went into the numbers used to calculate how much of one color to purge to get to another, and they modified the base numbers.

For example, here are the flushing numbers for a black/orange/green pumpkin I was printing for Halloween using the previous flushing values in version 1.9:
![Pumpkin Flushing Volumes 1 9](https://github.com/user-attachments/assets/3d706d0f-acfc-4921-aafc-55ebbe638579)
Here are those same values in version 1.10 BETA2 (I'm presuming they haven't changed from BETA to RTM):
![Pumpkin Flushing Volumes 1 10 BETA](https://github.com/user-attachments/assets/d9b0cb9c-b6f3-441a-8a9f-085b7d973d44)

Here are the same numbers but with a 50% flushing volumes multiplier (that used to work well for me using LeonFisherSkipper's long retraction Gcode) - v1.9 is first:
![Pumpkin Flushing Volumes 1 9 50percent](https://github.com/user-attachments/assets/0ea96d6a-df1b-4051-ac89-b0b18929b376)
![Pumpkin Flushing Volumes 1 10 BETA 50percent](https://github.com/user-attachments/assets/7c2c0ad0-05af-4902-80a3-a4683ecfd7f2)

My personal experience is these reduced values mean the previous .5 multiplier I used to use is no longer a good "standard" multiplier to use with long retraction as I had bleed from black to orange that I did not have previously.
